tests/src/Security/PermissionSet/Repository/PermissionSetTest.php

```diff
@@ -61,4 +63,150 @@ public function testSaving()
 
 		self::assertEquals($savedPermissionSet, $permissionSetSavedSelected);
 	}
+
+	/**
+	 * Asserts that the actual permissionset is equal to the expected permissionset
+	 *   --> It skips the "id" fields
+	 *
+	 * @param PermissionSets $expected
+	 * @param PermissionSets $actual
+	 */
+	public static function assertEqualPermissionSets(PermissionSets $expected, PermissionSets $actual)
+	{
+		self::assertEquals($expected->count(), $actual->count(), 'PermissionSets not even ' . PHP_EOL . 'expected: ' . print_r($expected, true) . 'actual: ' . print_r($actual, true));
+
+		foreach ($expected as $outputPermissionSet) {
+			self::assertCount(1, $actual->filter(function (PermissionSet $actualPermissionSet) use ($outputPermissionSet) {
+				return (
+					$actualPermissionSet->uid == $outputPermissionSet->uid &&
+					$actualPermissionSet->allow_cid == $outputPermissionSet->allow_cid &&
+					$actualPermissionSet->allow_gid == $outputPermissionSet->allow_gid &&
+					$actualPermissionSet->deny_cid == $outputPermissionSet->deny_cid &&
+					$actualPermissionSet->deny_gid == $outputPermissionSet->deny_gid
+				);
+			}), 'PermissionSet not found: ' . print_r($outputPermissionSet, true));
+		}
+	}
+
+	public function dataSet()
+	{
+		return [
+			'standard' => [
+				'permissionSets' => [
+					[
+						'uid'       => 42,
+						'allow_cid' => '<<43>>',
+						'allow_gid' => '',
+						'deny_cid'  => '<<44>>',
+						'deny_gid'  => '',
+					],
+					[
+						'uid'       => 42,
+						'allow_cid' => '',
+						'allow_gid' => '<<>>',
+						'deny_cid'  => '',
+						'deny_gid'  => '',
+					],
+					[
+						'uid'       => 42,
+						'allow_cid' => '<<44>>',
+						'allow_gid' => '',
+						'deny_cid'  => '',
+						'deny_gid'  => '',
+					],
+				],
+				'assertions' => [
+					[
+						'input' => [
+							'cid' => 43,
+							'uid' => 42,
+						],
+						'output' => new PermissionSets([
+							new PermissionSet(42, [43], [], [44], []),
+							new PermissionSet(42, [], [], [], []),
+						]),
+					],
+					[
+						'input' => [
+							'cid' => 44,
+							'uid' => 42,
+						],
+						'output' => new PermissionSets([
+							new PermissionSet(42, [], [], [], []),
```

Comment:
@MrPetovan - Is it intended that the output for `cid => 44` wouldn't return the PermissionSet with `deny_cid => [44]`: 
https://github.com/friendica/friendica/pull/10944/commits/dcbf8e18f0a7e7c0dafa2ae42ab70d0352ee7f68#diff-7d7b58f12132690f1a565de40abbdab31dd28e5c1afff2ee9b74471ff6ccff0cR112

Reply:
I added the test without it, but I'm not sure if it's right

Reply:
There should only be a single set of chevrons, like `<44>`